app/main/views/suppliers.py

```diff
@@ -810,12 +811,14 @@ def create_application_from_supplier(code):
 
     # hotfix for exception. shouldn't need to do this
     supplier.data = supplier.data or {}
+    application_type = application_type or 'upgrade'
 
-    if 'application_id' in supplier.data:
-        application = Application.query.filter(
-            Application.id == supplier.data['application_id']
-        ).first()
-        return jsonify(application=application.serializable)
+    existing_application = Application.query.filter(
+        Application.supplier_code == supplier.code,
+        or_(Application.status == 'submitted', Application.status == 'saved')
```

Comment:
rejected is like approved or deleted - I would consider all these statuses 'complete' with no editing in process